packages/flutter_tools/lib/executable.dart

```diff
@@ -179,8 +179,11 @@ Future<Null> _exit(int code) async {
   logger.flush();
 
   // Give the task / timer queue one cycle through before we hard exit.
-  await Timer.run(() {
+  Completer<Null> completer = new Completer<Null>();
+  Timer.run(() {
```

Comment:
looks good as is, but you could also change to `await new Future<Null>(() { ...`